internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestPushJobWithNativeReplication.java

```diff
@@ -255,9 +258,15 @@ public void testNativeReplicationForBatchPush(int recordCount, int partitionCoun
             }
 
             String pushJobDetailsStoreName = VeniceSystemStoreUtils.getPushJobDetailsStoreName();
-            Assert.assertEquals(
-                parentControllerClient.getAllValueSchema(pushJobDetailsStoreName).getSchemas().length,
-                AvroProtocolDefinition.PUSH_JOB_DETAILS.currentProtocolVersion.get().intValue());
+            waitForNonDeterministicAssertion(5, TimeUnit.SECONDS, () -> {
```

Comment:
Oh just to double check: Is 5 seconds enough to get through the flakiness?

Reply:
Although it is not clear to me why schema fetching for the system store will fail at this phase. Maybe it is "async" and even user store is done the system store is not yet...

Reply:
IDK if it is enough. The main thing I'm interested in for this part is the improved assertion error messages (especially the one from `assertCommand`...).

Reply:
Sure. We can give it a try. If that does not work I think there might be something async and not playing well.